### PR TITLE
[Backport 5.2] Updating an code host config maintains the correct unrestricted…

### PR DESCRIPTION
**Pull request by pjlast**
Fixes an issue where the logic that determines whether or not a code host connection is unrestricted differs between the code host connection creation and update code.

## Test plan

Tests updated

cc @sourcegraph/release-guild

### CODE REVIEW

File: internal/database/external_services.go [hunk 1]
```diff
@@ -1652,11 +1653,8 @@ WHERE EXISTS(
 	return v && exists, nil
 }
 
-// recalculateFields updates the value of the external service fields that are
-// calculated depending on the external service configuration, namely
-// `Unrestricted` and `HasWebhooks`.
-func (e *externalServiceStore) recalculateFields(es *types.ExternalService, rawConfig string) error {
-	es.Unrestricted = !envvar.SourcegraphDotComMode() && !gjson.Get(rawConfig, "authorization").Exists()
+func calcUnrestricted(config string) bool {
```
Comment (keegancsmith) [inline internal/database/external_services.go:1656]: I know this is a backport, so treat this as a driveby followup. But the mutliple checks on dotcom make this a little tricky to read. Seems like you could do something like "if dotcom return true" and then the rest of the logic may also be easier to read.

**Review by keegancsmith — APPROVED** [review]

**Comment by keegancsmith** [comment]
CI is not pleased.

File: internal/database/external_services.go (hunk 1 above)
Comment (pjlast) [reply internal/database/external_services.go:1656]: Yeah gotcha, thanks! I'll make a follow-up!

**Comment by pjlast** [comment]
@keegancsmith I see in the test I added I depended on a new field that wasn't backported. Updated the test, so 🤞

**Pull request merged**